pkg/controller/instance/instance_controller.go

```diff
@@ -115,295 +76,201 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// ---------- 1. Query the current state ----------
 
-			// Identify plan to be executed by this change.
-			var planName string
-			var planFound bool
-
-			if old.Spec.OperatorVersion != new.Spec.OperatorVersion {
-				// It's an upgrade!
-				names := []string{"upgrade", "update", "deploy"}
-				for _, n := range names {
-					if _, planFound = ov.Spec.Plans[n]; planFound {
-						planName = n
-						break
-					}
-				}
+	log.Printf("InstanceController: Received Reconcile request for instance \"%+v\"", request.Name)
+	instance, err := r.getInstance(request)
+	if err != nil {
+		if apierrors.IsNotFound(err) { // not retrying if instance not found, probably someone manually removed it?
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
 
-				if !planFound {
-					log.Printf("InstanceController: Could not find any plan to use to upgrade instance %v", new.Name)
-					return false
-				}
-			} else if !reflect.DeepEqual(old.Spec, new.Spec) {
-				for k := range parameterDifference(old.Spec.Parameters, new.Spec.Parameters) {
-					// Find the spec of the updated parameter.
-					paramFound := false
-					for _, param := range ov.Spec.Parameters {
-						if param.Name == k {
-							paramFound = true
-
-							if param.Trigger != "" {
-								planName = param.Trigger
-								planFound = true
-							}
-
-							break
-						}
-					}
-
-					if paramFound {
-						if !planFound {
-							// The parameter doesn't have a trigger, try to find the corresponding default plan.
-							names := []string{"update", "deploy"}
-							for _, n := range names {
-								if _, planFound = ov.Spec.Plans[n]; planFound {
-									planName = n
-									planFound = true
-									break
-								}
-							}
-
-							if planFound {
-								log.Printf("InstanceController: Instance %v updated parameter %v, but it is not associated to a trigger. Using default plan %v\n", new.Name, k, planName)
-							}
-						}
-
-						if !planFound {
-							log.Printf("InstanceController: Could not find any plan to use to update instance %v", new.Name)
-						}
-					} else {
-						log.Printf("InstanceController: Instance %v updated parameter %v, but parameter not found in operatorversion %v\n", new.Name, k, ov.Name)
-					}
-				}
-			} else {
-				// FIXME: reading the status here feels very wrong, and so does the fact
-				// that this predicate funcion has side effects. We could probably move
-				// all this logic to the reconciler if we stored the parameters in the
-				// `PlanExecution`.
-				// See https://github.com/kudobuilder/kudo/issues/422
-				if new.Status.ActivePlan.Name == "" {
-					log.Printf("InstanceController: Old and new spec matched...\n %+v ?= %+v\n", old.Spec, new.Spec)
-					planName = "deploy"
-					planFound = true
-				}
-			}
+	ov, err := r.getOperatorVersion(instance)
+	if err != nil {
+		return reconcile.Result{}, err // OV not found has to be retried because it can really have been created after I
```

Comment:
the comment ends oddly... "after I"?  is :instance" meant?

Reply:
yes, fixed